app/controllers/vacancies_controller.rb

```diff
@@ -24,7 +24,14 @@ def index
   end
 
   def show
-    vacancy = Vacancy.listed.friendly.find(id)
+    begin
+      vacancy = Vacancy.listed.friendly.find(id)
+    rescue ActiveRecord::RecordNotFound
+      raise unless Vacancy.trashed.friendly.exists?(id)
```

Comment:
I like the implementation, but can we get some tests for this?

Reply:
You can probably use Rspec's [render template matcher](https://relishapp.com/rspec/rspec-rails/docs/matchers/render-template-matcher) in a controller test for this